src/backtrace_internal.h

```diff
@@ -0,0 +1,43 @@
+#ifndef __LIBDRAGON_BACKTRACE_INTERNAL_H
+#define __LIBDRAGON_BACKTRACE_INTERNAL_H
+
+/** @brief The "type" of funciton as categorized by the backtrace heuristic (__bt_analyze_func) */
+typedef enum {
+    BT_FUNCTION,                ///< Regular function with a stack frame
+    BT_FUNCTION_FRAMEPOINTER,   ///< The function uses the register fp as frame pointer (normally, this happens only when the function uses alloca)
+    BT_EXCEPTION,               ///< This is an exception handler (inthandler.S)
+    BT_LEAF                     ///< Leaf function (no calls), no stack frame allocated, sp/ra not modified
```

Comment:
Tail-calling is a property of how you call a function, it's not a property of the function itself. Any function of the above types might tailcall another function of any of the above types.

Reply:
Actually what I meant was a recursive tailcall. In that case it will just call itself without pushing anything to the stack. So shouldn't be any different than a leaf node IINM?

Reply:
Yes. In general a recursive tail call is handled by the compiler transforming the code in a loop, so at the assembly level you can't really tell it was written as recursive tail call. That's because a real tail call means popping the whole stack frame and then jumping, but in the recursive case it would be a waste of time to pop and push the same stack frame.